cms/djangoapps/contentstore/signals/handlers.py

```diff
@@ -43,6 +51,48 @@ def wrapper(*args, **kwargs):
     return task_decorator
 
 
+def create_catalog_data_for_signal(course_key: CourseKey) -> Optional[CourseCatalogData]:
+    """
+    Creates data for catalog-info-changed signal when course is published.
+
+    Arguments:
+        course_key: Key of the course to announce catalog info changes for
+
+    Returns:
+        Data for signal, or None if not appropriate to send on this signal.
+    """
+    # Only operate on real courses, not libraries.
+    if not course_key.is_course:
+        return None
+
+    store = modulestore()
+    with store.branch_setting(ModuleStoreEnum.Branch.published_only, course_key):
+        course = store.get_course(course_key)
+
+        return CourseCatalogData(
+            course_key=course_key.for_branch(None),
```

Comment:
I don't think `for_branch` is necessary here.

Reply:
Is that because `course_published` already does it, or because some other reason?

Reply:
I think this is actually related to your comment below about how you're worried that we're getting course keys with branch versions passed around. That's not really common, but we're in modulestore land, so who knows.